### PR TITLE
minor changes to adapt to mf

### PR DESCRIPTION
**Pull request by nikita-0209**
creating a draft PR because I keep forgetting the names of branches I am working on

### CODE REVIEW

File: gflownet/envs/grid.py [hunk 1]
```diff
@@ -348,6 +349,8 @@ def get_all_terminating_states(self) -> List[List]:
         return all_x.tolist()
 
     def get_uniform_terminating_states(self, n_states: int, seed: int) -> List[List]:
+        # To Discuss: don't we want this seed to be the same as the one set in main()
```
Comment (alexhernandezgarcia) [inline gflownet/envs/grid.py:352]: Yes

File: gflownet/envs/grid.py [hunk 2]
```diff
@@ -58,9 +59,12 @@ def __init__(
...
             self.statetorch2proxy = self.statetorch2oracle
-        # Set up proxy
-        self.proxy.n_dim = self.n_dim
-        self.proxy.setup()
+        # Set up oracle
+        self.oracle.n_dim = self.n_dim
+        self.oracle.setup()
+        # TODO: change to something more general. this would be invalid if num_fid>5
+        self.invalid_action = self.n_dim + 5
+        self.proxy_factor = 1.0
```
Comment (nikita-0209) [inline gflownet/envs/grid.py:67]: need to make this dependant on proxy not oracle

File: gflownet/envs/base.py [hunk 3]
```diff
@@ -58,10 +59,13 @@ def __init__(
             self.oracle = self.proxy
         else:
             self.oracle = oracle
-        if self.oracle.higher_is_better:
-            self.proxy_factor = 1.0
-        else:
-            self.proxy_factor = -1.0
+            # higher_is_better is dependant on proxy
+            # but proxy is set after env is iniatiliased
+            # TODO: need to think about how to do iniatilise higherIsBetter
```
Comment (alexhernandezgarcia) [inline gflownet/envs/base.py:64]: I don't understand the problem. The proxy is initialized (in `main.py`) before the environment is actually passed as an argument to the environment. So what's the problem of taking `higher_is_better` from the proxy?

File: gflownet/envs/base.py [hunk 4]
```diff
@@ -700,7 +707,8 @@ def make_data_set(self, config):
             samples = self.env.get_uniform_terminating_states(config.n, config.seed)
         else:
             return None, None
-        energies = self.env.oracle(self.env.statebatch2oracle(samples)).tolist()
+        energies = self.env.proxy(self.env.statebatch2proxy(samples)).tolist()
+        # energies = self.env.oracle(self.env.statebatch2oracle(samples)).tolist()
```
Comment (alexhernandezgarcia) [inline gflownet/envs/base.py:711]: Do you really want to keep these two lines, one commented?

File: gflownet/envs/grid.py (hunk 2 above)
Comment (alexhernandezgarcia) [reply gflownet/envs/grid.py:68]: Your comment is to explain your changes? If so, I agree.

File: gflownet/envs/grid.py [hunk 5]
```diff
@@ -347,7 +352,57 @@ def get_all_terminating_states(self) -> List[List]:
         )
         return all_x.tolist()
 
-    def get_uniform_terminating_states(self, n_states: int, seed: int) -> List[List]:
-        rng = np.random.default_rng(seed)
-        states = rng.integers(low=0, high=self.length, size=(n_states, self.n_dim))
+    def get_uniform_terminating_states(self, n_states: int) -> List[List]:
+        states = np.random.randint(low=0, high=self.length, size=(n_states, self.n_dim))
+        # To Discuss: can we return a tensor?
```
Comment (alexhernandezgarcia) [inline gflownet/envs/grid.py:357]: Yes, but making sure that nothing breaks elsewhere ^^

File: gflownet/envs/grid.py [hunk 6]
```diff
@@ -347,7 +352,57 @@ def get_all_terminating_states(self) -> List[List]:
...
+        index = states.long().detach().cpu().numpy()
+        grid_scores[index[:, 0], index[:, 1]] = scores
+        im = ax.imshow(grid_scores)
+        divider = make_axes_locatable(ax)
+        ax.set_title(figure_title)
+        cax = divider.append_axes("right", size="5%", pad=0.05)
+        plt.colorbar(im, cax=cax)
+        plt.show()
+        plt.close()
+        return fig
```
Comment (alexhernandezgarcia) [inline gflownet/envs/grid.py:408]: Thanks so much for adding all this!

File: gflownet/gflownet.py [hunk 7]
```diff
@@ -855,6 +862,7 @@ def test(self):
                 hist[tuple(x)] += 1
             z_pred = sum([hist[tuple(x)] for x in x_tt]) + 1e-9
             density_pred = np.array([hist[tuple(x)] / z_pred for x in x_tt])
+            corr = np.corrcoef(density_pred, density_true)
```
Comment (alexhernandezgarcia) [inline gflownet/gflownet.py:865]: This correlation is not the "old" correlation between logq and energy. That was not your intention was it?

Comment (nikita-0209) [reply gflownet/gflownet.py:865]: Yep it is not the old one. I just wanted an easy measure of the correlation between the frequency of states generated by the sampler (probability of sampling a state) vs the state's reward.

File: gflownet/envs/base.py (hunk 3 above)
Comment (nikita-0209) [reply gflownet/envs/base.py]: In the active learning code, the proxy (acquisition function) is initialized after the env as the acquisition function is dependant on the regressor (MLP), which is dependant on the env. I've added a `set_proxy()` function to take care of this
